src/dpki.rs

```diff
@@ -0,0 +1,377 @@
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&format!("{}", self))
+    }
+}
+
+impl<'d> Deserialize<'d> for SigningPublicKey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'d>,
+    {
+        let hcid = String::deserialize(deserializer)?;
+        Self::from_bytes(
+            &hcid::HcidEncoding::with_kind("hcs0")
+                .map_err(de::Error::custom)?
+                .decode(&hcid)
+                .map_err(de::Error::custom)?
+        ).map_err(de::Error::custom)
+    }
+}
+
+
+// A regular 256-bit seed is encoded in 24 BIP39 words
+#[derive(Debug)]
+pub struct Seed(pub [u8; SEED_SIZE]);
+
+// An encrypted seed adds a 256-bit authentication MAC + salt, and is encoded in 48 BIP39 words.  In
+// holochain_dpki::seed::MnemonicableSeed for EncryptedSeed, we see that a all-0 nonce is used in
+// the aead encryption, and the Seed ciphertext is first, followed by the authentication MAC, and
+// finally the random salt.5
+pub struct EncryptedSeed(pub [u8; ENCRYPTED_SEED_SIZE]);
+
+/*
+ * from holochain_dpki::seed
+ *
+pub(crate) fn pw_enc_zero_nonce(
+    data: &[u8],
+    passphrase: &mut SecBuf,
+    config: Option<PwHashConfig>,
+) -> HcResult<EncryptedData> {
+    let mut salt = SecBuf::with_insecure(SALTBYTES);
+    salt.randomize();
+    let mut nonce = SecBuf::with_insecure(NONCEBYTES);
+    nonce.write(0, &[0; NONCEBYTES])?;
+    let data = pw_enc_base(data, passphrase, &mut salt, &mut nonce, config)?;
+    Ok(data)
+}
+
+/// Private general wrapper of pw_enc
+fn pw_enc_base(
+    data: &mut SecBuf,
+    passphrase: &mut SecBuf,
+    mut salt: &mut SecBuf,
+    mut nonce: &mut SecBuf,
+    config: Option<PwHashConfig>,
+) -> HcResult<EncryptedData> {
+    let mut secret = SecBuf::with_secure(kx::SESSIONKEYBYTES);
+    let mut cipher = SecBuf::with_insecure(data.len() + aead::ABYTES);
+    pw_hash(passphrase, &mut salt, &mut secret, config)?;
+    aead::enc(data, &mut secret, None, &mut nonce, &mut cipher)?;
+
+    let salt = salt.read_lock().to_vec();
+    let nonce = nonce.read_lock().to_vec();
+    let cipher = cipher.read_lock().to_vec();
+    // Done
+    Ok(EncryptedData {
+        salt,
+        nonce,
+        cipher,
+    })
+}
+ */
+
+pub trait SeedData
+where
+    Self: Sized,
+{
+    fn from_bytes(b: &[u8]) -> Result<Self, Error>;
+    fn as_bytes<'a>(&'a self) -> &'a [u8];
+}
+
+impl SeedData for Seed {
+    fn from_bytes(b: &[u8]) -> Result<Self, Error> {
+        ensure!(b.len() == SEED_SIZE, "Incorrect length for Seed");
+        let mut seed_bytes = [0u8; SEED_SIZE];
+        seed_bytes.copy_from_slice(&b[..SEED_SIZE]);        
+        Ok(Self(seed_bytes))
+    }
+    fn as_bytes<'a>(&'a self) -> &'a [u8] {
+        &self.0
+    }
+}
+
+impl Seed {
+    pub fn encrypt(
+        &self,
+        _password: &str, // TODO: implmenent aead encryption!
```

Comment:
Holochain should use passphrase manager to decrypt seed, rather than `holo-config-derive`.